modules/swagger-parser-v3/src/main/java/io/swagger/parser/v3/util/OpenAPIDeserializer.java

```diff
@@ -1,23 +1,2081 @@
+        if(node != null) {
+            components.setLinks(getLinks(node,location,result));
+        }
+
+        node = getObject("callbacks",obj,false,location,result);
+        if(node != null) {
+            components.setCallbacks(getCallbacks(node,location,result));
+        }
+        components.setExtensions(new LinkedHashMap<>());
+
+        Set<String> keys = getKeys(obj);
+        for(String key : keys) {
+            if(key.startsWith("x-")) {
+                components.getExtensions().put(key, Json.mapper().convertValue(obj.get(key), Object.class));
+            }
+        }
+
+        return  components;
+    }
+
+    public List<Tag> getTagList(ArrayNode obj, String location, ParseResult result) {
+        if (obj == null) {
+            return null;
+        }
+        List<Tag> tags = new ArrayList<>();
+        for (JsonNode item : obj) {
+            if (item.getNodeType().equals(JsonNodeType.OBJECT)) {
+                Tag tag = getTag((ObjectNode) item, location, result);
+                if (tag != null) {
+                    tags.add(tag);
+                }
+            }
+        }
+        return tags;
+    }
+
+    public Tag getTag(ObjectNode obj, String location, ParseResult result) {
+        if (obj == null) {
+            return null;
+        }
+
+        Tag tag = new Tag();
+
+        String value = getString("name", obj, true, location, result);
+        tag.setName(value);
+
+        value = getString("description", obj, false, location, result);
+        tag.setDescription(value);
+
+        Set<String> tagKeys = getKeys(obj);
+        for (String tagName : tagKeys) {
+            JsonNode tagValue = obj.get(tagName);
+            if (!tagValue.getNodeType().equals(JsonNodeType.OBJECT)) {
+                result.invalidType(location, tagName, "object", tagValue);
+            } else {
+                ObjectNode tagObj = (ObjectNode) tagValue;
+                ExternalDocumentation externalDocs = getExternalDocs(tagObj, String.format("%s.%s'", location, tagName), result);
+                tag.setExternalDocs(externalDocs);
+            }
+        }
+
+
+        return tag;
+    }
+
+
+
+    public List<Server> getServersList(ArrayNode obj, String location, ParseResult result) {
+
+        List<Server> servers = new ArrayList<>();
+        if (obj == null) {
+            return null;
+
+        }
+        for (JsonNode item : obj) {
+            if (item.getNodeType().equals(JsonNodeType.OBJECT)) {
+                Server server = getServer((ObjectNode) item, location, result);
+                if (server != null) {
+                    servers.add(server);
+                }
+            }
+        }
+        return servers;
+    }
+
+    public Server getServer(ObjectNode obj, String location, ParseResult result) {
+        if (obj == null) {
+            return null;
+        }
+
+        Server server = new Server();
+
+        String value = getString("url", obj, true, location, result);
```

Comment:
use the `ParseResult` and `location` to store messages back to the user.  In this case, `url` is a mandatory field and if it's not present, you can tell the user that it's missing